src/app/server/BUILD.gn

```diff
@@ -63,6 +64,13 @@ static_library("server") {
     "Server.h",
   ]
 
+  if (chip_device_config_enable_joint_fabric) {
+    sources += [
+      "JointFabricDatastore.cpp",
+      "JointFabricDatastore.h",
```

Comment:
Don't these need to get removed from the unconditional list above?

Reply:
The changes have been moved under chip_device_config_enable_joint_fabric flag

Reply:
I see #38842 got through without unit testing.

Please apply these changes:
  - [ ] use a source_set for these 2 files (so it can be referenced individually)
  - [ ] add unit tests in a src/app/server/tests . Reference the source set so you can write the unit test